src/node/txreconciliation.cpp

```diff
@@ -121,26 +194,233 @@ class TxReconciliationTracker::Impl
 
         const uint256 full_salt{ComputeSalt(local_salt, remote_salt)};
         recon_state->second = TxReconciliationState(!is_peer_inbound, full_salt.GetUint64(0), full_salt.GetUint64(1));
+        if (!is_peer_inbound) m_queue.push_back(peer_id);
         return ReconciliationRegisterResult::SUCCESS;
     }
 
+    size_t AddToSet(NodeId peer_id, const std::vector<uint256>& txs_to_reconcile) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        Assume(txs_to_reconcile.size() > 0);
+        LOCK(m_txreconciliation_mutex);
+        if (!IsPeerRegistered(peer_id)) return 0;
+        auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+
+        size_t added = 0;
+        for (auto& wtxid : txs_to_reconcile) {
+            added += recon_state.m_local_set.insert(wtxid).second;
```

Comment:
Correct the max set size (`MAX_SET_SIZE`) is enforced by `ShouldFanoutTo()` which happens before the call to `AddToSet()`, don't know if it could be more conservative with another check here.